docs/modules/ROOT/pages/reference/architecture.adoc

```diff
@@ -60,6 +60,10 @@ For each dependency which is stored in this repository, Commodore ensures that t
 Regardless of the value of key `path`, Commodore creates a checkout of the  complete repository in `dependencies/<dependency-name>`.
 However, Commodore will create a symlink to the specified path when making the dependency available in the hierarchy.
 
+For components which are instantiated multiple times, Commodore ensures that an additional Git worktree in `dependencies/<instance-name>` exists for each component instance and is checked out to the instance's desired component version.
```

Comment:
I quickly added this section since we now create a Git worktree for each component instance regardless of whether a component supports per-instance versions.